app/controllers/application_controller.rb

```diff
@@ -1,10 +1,133 @@
 require './config/environment'
 
+# ApplicationController handles requests for homepage, signup and logins
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+  get '/' do
+    redirect "/tweets" if Helpers.is_logged_in?(session)
+    erb :index
+  end
+
+  get '/signup' do
+    redirect "tweets" if Helpers.is_logged_in?(session)
+    erb :"users/signup"
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect "/signup", locals: {message: "Please submit enter all the information to complete signup."}
+    else
+      @user = User.create(
+        username: params[:username],
+        email: params[:email],
+        password: params[:password]
+      )
+      session[:user_id] = @user.id
+      redirect "/tweets", locals: {message: "Welcome to Fwitter. Thank you for signing up."}
```

Comment:
Wait what error did you get? I cloned my repo and was able to make a new
account.

On Sun, Feb 28, 2016 at 6:31 PM, Randall Reed <notifications@github.com>
wrote:

> In app/controllers/application_controller.rb
> <https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/24#discussion_r54359031>
> :
>
> > +  get '/signup' do
> > +    redirect "tweets" if Helpers.is_logged_in?(session)
> > +    erb :"users/signup"
> > +  end
> > +
> > +  post '/signup' do
> > +    if params[:username] == "" || params[:email] == "" || params[:password] == ""
> > +      redirect "/signup", locals: {message: "Please submit enter all the information to complete signup."}
> > +    else
> > +      @user = User.create(
> > +        username: params[:username],
> > +        email: params[:email],
> > +        password: params[:password]
> > +      )
> > +      session[:user_id] = @user.id
> > +      redirect "/tweets", locals: {message: "Welcome to Fwitter. Thank you for signing up."}
>
> I got an error when trying to signup. Is it still working for you?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/24/files#r54359031>
> .
>